maths/series/arithmetico_geometric_sequence.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+"""
+This is a pure Python implementation of the Arithmetic–Geometric Sequence algorithm
+https://en.wikipedia.org/wiki/Arithmetico%E2%80%93geometric_sequence
+For doctests run following command:
+python -m doctest -v arithmetic_geometric_sequence.py
+or
+python3 -m doctest -v arithmetic_geometric_sequence.py
+For manual testing run:
+python3 arithmetic_geometric_sequence.py
+"""
+
+
+class AGS:
+    def __init__(self, a, d, b, r, n):
+        self.a = a
+        self.d = d
+        self.b = b
+        self.r = r
+        self.n = n
+
+    """Pure Python implementation of Arithmetic–Geometric Sequence algorithm
+    :params a: Initial Value For Arithmetic Progression
+            d: Difference For Arithmetic Progression
+            b: Initial Value For Geometric Progression
+            r: Common Ratio For Geometric Progression
+            n: Number Of Terms
+            k: Value Of K In K-th Term
+    Examples:
+    Enter Initial Value For A.P. : 2
+    Enter Difference For A.P. : 3
+    Enter Initial Value For G.P. : 4
+    Enter Common Ratio For G.P. : 0.5
+    Enter Number Of Terms : 5
+    Full Series :
+    ['2.0 x 4.0', '(2.0 + 3.0) x 4.0x0.5^1', '(2.0 + 6.0) x 4.0x0.5^2',
+    '(2.0 + 9.0) x 4.0x0.5^3', '(2.0 + 12.0) x 4.0x0.5^4']
+    Value Of Last Term : 3.50
+    Sum Of Your A.G.S. : 35.00
+    Infinite Series Sum : 40.00
+    Value Of Which Term You Want : 2
+    Value Of 2-th Term : 10.00
+    """
+
+    def full_series(self):
+        series = []
+        for i in range(self.n):
+            if i == 0:
+                series.append(str(self.a) + " x " + str(self.b))
+            else:
+                ap_part = str(self.a) + " + " + str(i * self.d)
+                gp_part = str(self.b) + "x" + str(self.r) + "^" + str(i)
+                series.append("(" + ap_part + ")" + " x " + gp_part)
```

Comment:
```suggestion
                series.append(f"({ap_part}) x {gp_part})
```

Reply:
f-strings as discussed in CONTRIBUTING.md.

Reply:
You haven't closed the f-string